app/src/main/java/campus/tech/kakao/map/ui/map/MapActivity.kt

```diff
@@ -94,21 +94,28 @@ class MapActivity : AppCompatActivity() {
             val longitude = data?.getDoubleExtra(EXTRA_PLACE_LONGITUDE, 0.0) ?: 0.0
             val latitude = data?.getDoubleExtra(EXTRA_PLACE_LATITUDE, 0.0) ?: 0.0
 
-            val newLocation = Location(name, latitude, longitude, address)
-            locationViewModel.saveLocation(newLocation)
-            setBottomSheet()
-            startMapView()
+            val newLocation = LocationDomain(name, latitude, longitude, address)
+
+            saveLocationAndWaitForUpdate(newLocation)
         }
     }
 
     /**
-     * bottom sheet의 name과 address의 text를 설정하는 함수.
+     * 위치를 저장하고 업데이트를 기다리는 함수.
      *
+     * @param newLocation 저장할 새로운 위치 객체.
      */
-    private fun setBottomSheet() {
-        locationViewModel.location.value.let { location ->
-            binding.bottomSheetPlaceNameTextView.text = location.name
-            binding.bottomSheetPlaceAddressTextView.text = location.address
+    private fun saveLocationAndWaitForUpdate(newLocation: LocationDomain) {
+        lifecycleScope.launch {
+            repeatOnLifecycle(Lifecycle.State.STARTED) {
+                locationViewModel.saveLocation(newLocation)
+                locationViewModel.location.collectLatest { location ->
+                    if (location == newLocation) {
+                        startMapView()
+                        cancel()
+                    }
+                }
```

Comment:
이런식으로 처리해보는건 어떨까요? cacnel 을 하기보다 take 1 으로 명확하게 하는 것이 좋을 것 같아요!

```suggestion
                locationViewModel.location
                    .filter { it == newLocation }
                    .take(1)
                    .collectLatest { location ->
                          startMapView()
                          cancel()
                }
```